src/player/sample_field_evaluator.h

```diff
@@ -31,27 +31,76 @@
 #include "predict_state.h"
 
 #include <vector>
+#include <map>
+
+#ifdef USE_GRPC
+#include "../grpc-generated/service.pb.h"
+using protos::PlannerEvalution;
+#endif
 
 namespace rcsc {
 class AbstractPlayerObject;
 class Vector2D;
+class WorldModel;
 }
 
 class ActionStatePair;
 
 class SampleFieldEvaluator
     : public FieldEvaluator {
 private:
-
+    std::vector< double > m_opponent_negetive_effect_by_distance;
+    std::vector< double > m_opponent_negetive_effect_by_reach_steps;
+
+    bool m_use_action_term = false;
```

Comment:
Why did you define the paramater here should we get it from the python code ? or is this a test only ?